app/screens/home/search/modifiers/modifier.tsx

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import {Text, View} from 'react-native';
+
+import CompassIcon from '@components/compass_icon';
+import MenuItem from '@components/menu_item';
+import {useTheme} from '@context/theme';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+export const MODIFIER_LABEL_HEIGHT = 48;
+
+const getStyleFromTheme = makeStyleSheetFromTheme((theme) => {
+    return {
+        container: {
+            flext: 1,
+            height: MODIFIER_LABEL_HEIGHT,
+            marginLeft: 20,
+            justifyContent: 'center',
+        },
+        modifierItemLabelContainer: {
+            flexDirection: 'row',
+            alignItems: 'center',
+            marginTop: -4,
+        },
+        icon: {
+            marginTop: 3,
+        },
+        modifierLabelValue: {
+            marginLeft: 16,
+            color: theme.centerChannelColor,
+            ...typography('Body', 200, 'SemiBold'),
+        },
+        modifierItemDescription: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 200, 'Regular'),
+        },
+    };
+});
+
+export type ModifierItem = {
+        description: string;
+        testID: string;
+        term: string;
+}
+
+type Props = {
+    item: ModifierItem;
+    setSearchValue: (value: string) => void;
+    searchValue?: string;
+}
+
+const Modifier = ({item, searchValue, setSearchValue}: Props) => {
+    const theme = useTheme();
+
+    const handlePress = useCallback(() => {
+        addModifierTerm(item.term);
+    }, [item.term, searchValue]);
+
+    const style = getStyleFromTheme(theme);
+
+    const addModifierTerm = preventDoubleTap((modifierTerm) => {
+        let newValue = '';
+        if (!searchValue) {
+            newValue = modifierTerm;
+        } else if (searchValue.endsWith(' ')) {
+            newValue = `${searchValue}${modifierTerm}`;
+        } else {
+            newValue = `${searchValue} ${modifierTerm}`;
+        }
+
+        setSearchValue(newValue);
+    });
+
+    return (
+        <MenuItem
```

Comment:
see my comment about `OptionItem`

Reply:
Heres the output using the `OptionItem` component.  I added an `inline` boolean prop which places the description inline with the label.  @matthewbirtch, the styling is slightly different than the figma design. @avinashlng1080, are you ok with my changes to `OptionItem` and the new `inline` prop?

@enahum, thanks for the heads up. This reduces the `Modifier` component code significantly.

Figma Design: 
<img width="402" alt="image" src="https://user-images.githubusercontent.com/7575921/176247763-b3d2e05b-3f65-49ea-9adc-04fbdb03d0a9.png">

Using OptionItem component:
<img width="514" alt="image" src="https://user-images.githubusercontent.com/7575921/176247600-b7c0940a-be19-485e-b484-8d47e6559b54.png">

Reply:
I guess we could pass additional styles to the `OptionItem` component to make it match the design or we can keep the keep MenuItem and then refactor once MenuItem is removed from the codebase

Reply:
I think passing styles to the `OptionItem` would be nice, but also allow alot of variabilty in it's look. 

I could also adjust the logic when `inline` is set:

* if inline === false:
  * use the default optionItem ServiceSettings
* if inline === true and description = '':
  * use default optionItem ServiceSettings
* if inline === true && description:
  * use output style in Figma.

Reply:
I can revert the change to use OptionItem, but I am 1/5 we keep it

Reply:
If we can keep the styles, that's ideal, but if it adds too much bloat to the code then I would be ok with the way it looks in your screenshot above @jfrerich

Reply:
I added some logic to the component that will render a one-line option as shown in the figma design. If the `inline` prop is true and `description` is not empty, the Option item text will render the same as desired in the Figma design.